File: Problem_Solving/Algorithms/Warmup/6_Plus_Minus/Solution.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    static void plusMinus(int[] arr) {
        double posNums=0.0;
        double negNums=0.0;
        double zeroNums=0.0;
        double posFraction=0.0;
        double negFraction=0.0;
        double zeroFraction=0.0;
        DecimalFormat df=new DecimalFormat("0.000000");
        df.setRoundingMode(RoundingMode.HALF_UP);
        int arrLen=arr.length;

        for(int i=0;i<arrLen;i++){
            if(arr[i]==0)
                zeroNums++;
            else if(arr[i]>0)
                posNums++;
            else
                negNums++;
        }

        posFraction=posNums/arrLen;
        negFraction=negNums/arrLen;
        zeroFraction=zeroNums/arrLen;

        System.out.println(df.format(posFraction) + "\n" + 
            df.format(negFraction) + "\n" + 
            df.format(zeroFraction));
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        int[] arr = new int[n];

        String[] arrItems = scanner.nextLine().split(" ");
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        for (int i = 0; i < n; i++) {
            int arrItem = Integer.parseInt(arrItems[i]);
            arr[i] = arrItem;
        }

        plusMinus(arr);

        scanner.close();
    }
}

File: Problem_Solving/Algorithms/Warmup/6_Plus_Minus/Solution.py
import math
import os
import random
import re
import sys

def plusMinus(arr):
    posNums=0.0
    negNums=0.0
    zeroNums=0.0
    posFraction=0.0
    negFraction=0.0
    zeroFraction=0.0
    arrLen=len(arr)

    for i in range(arrLen):
        if arr[i]==0:
            zeroNums+=1
        elif arr[i]>0:
            posNums+=1
        else:
            negNums+=1

    posFraction=posNums/arrLen
    negFraction=negNums/arrLen
    zeroFraction=zeroNums/arrLen
    
    print(f'{posFraction:.6f}\n{negFraction:.6f}\n{zeroFraction:.6f}') 

if __name__ == '__main__':
    n = int(input())

    arr = list(map(int, input().rstrip().split()))

    plusMinus(arr)
